packages/flutter/lib/src/cupertino/text_selection_toolbar.dart

```diff
@@ -0,0 +1,1028 @@
+/// text.
+///
+/// Tries to position itself above [anchorAbove], but if it doesn't fit, then
+/// it positions itself below [anchorBelow].
+///
+/// If any children don't fit in the menu, an overflow menu will automatically
+/// be created.
+///
+/// See also:
+///
+///  * [TextSelectionControls.buildToolbar], where this is used by default to
+///    build an iOS-style toolbar.
+///  * [TextSelectionToolbar], which is similar, but builds an Android-style
+///    toolbar.
+class CupertinoTextSelectionToolbar extends StatelessWidget {
+  /// Creates an instance of CupertinoTextSelectionToolbar.
+  const CupertinoTextSelectionToolbar({
+    Key? key,
+    required this.anchorAbove,
+    required this.anchorBelow,
+    required this.children,
+    this.toolbarBuilder = _defaultToolbarBuilder,
+  }) : assert(children.length > 0),
+       super(key: key);
+
+  /// {@macro flutter.material.TextSelectionToolbar.anchorAbove}
+  final Offset anchorAbove;
+
+  /// {@macro flutter.material.TextSelectionToolbar.anchorBelow}
+  final Offset anchorBelow;
+
+  /// {@macro flutter.material.TextSelectionToolbar.children}
+  ///
+  /// See also:
+  ///   * [CupertinoTextSelectionToolbarButton], which builds a default
+  ///     Cupertino-style text selection toolbar text button.
+  final List<Widget> children;
+
+  /// {@macro flutter.material.TextSelectionToolbar.toolbarBuilder}
+  ///
+  /// The given anchor and isAbove can be used to position an arrow, as in the
+  /// default Cupertino toolbar.
+  final CupertinoToolbarBuilder toolbarBuilder;
+
+  // Builds a toolbar just like the default iOS toolbar, with the right color
+  // background and a rounded cutout with an arrow.
+  static Widget _defaultToolbarBuilder(BuildContext context, Offset anchor, bool isAbove, Widget child) {
+    return _CupertinoTextSelectionToolbarShape(
+      anchor: anchor,
+      isAbove: isAbove,
+      child: DecoratedBox(
+        decoration: const BoxDecoration(color: _kToolbarDividerColor),
+        child: child,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMediaQuery(context));
+    final MediaQueryData mediaQuery = MediaQuery.of(context);
+
+    final double paddingAbove = mediaQuery.padding.top + _kToolbarScreenPadding;
+    final double toolbarHeightNeeded = paddingAbove
+        + _kToolbarContentDistance
+        + _kToolbarHeight;
+    final bool fitsAbove = anchorAbove.dy >= toolbarHeightNeeded;
+
+    const Offset contentPaddingAdjustment = Offset(0.0, _kToolbarContentDistance);
+    final Offset localAdjustment = Offset(_kToolbarScreenPadding, paddingAbove);
+
+    return Padding(
+      padding: EdgeInsets.fromLTRB(
+        _kToolbarScreenPadding,
+        paddingAbove,
+        _kToolbarScreenPadding,
+        _kToolbarScreenPadding,
+      ),
+      child: CustomSingleChildLayout(
+        delegate: TextSelectionToolbarLayoutDelegate(
```

Comment:
This logic is now shared with Material via TextSelectionToolbarLayoutDelegate.  I deleted the duplicate logic that did this for Cupertino.